tests/components/threshold/test_binary_sensor.py

```diff
@@ -16,7 +16,24 @@
 from tests.common import MockConfigEntry
 
 
-async def test_sensor_upper(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("from_val", "to_val", "expected_position", "expected_state"),
+    [
+        (None, 15, "below", "off"),  # at threshold
+        (15, 16, "above", "on"),
+        (16, 14, "below", "off"),
+        (14, 15, "below", "off"),
```

Comment:
Now that you touch this file it is probably nice to use some more constants for all strings in here, like STATE_OFF, STATE_ON, STATE_UNKNOWN

Reply:
Done.